### ISSUE TITLE
Update to Next.js 12.x features

### ISSUE THREAD

**Issue by francinelucca** [carbon-design-system/carbon-platform#841]
**Summary**

We're waiting on next to release v12.1.7 so we can get some bug fixes in:

- package.json error on npm install
- move <Script/> from _app.js to _document.js

when this new version is released, let's upgrade and make the necessary changes and verify package.json error is no more

**Job stories**

What value will this deliver?

1. As a developer, I want to declare my Script tags inside my _document.js file, so I can be in compliance with next standards and have a more structurally sound application.
2. As a developer, I don't want to run into errors when doing an npm install

**Acceptance criteria**

What requirements must be met to complete this request?

1. Package.json error is not present anymore (if it is, let's open another ticket or reopen the old one)
3. Scripts tags are now located in _document.js file instead of _app.js and commonjs script still loads as expected

**Comment by francinelucca** [comment]
@jdharvey-ibm tagging you in case you want to add a better explanation to the packagejson error 😅

**Comment by mattrosno** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Development-624c3fc8b7972e001130fef7/issues/carbon-design-system/carbon-platform/841?planning-poker) with ZenHub @alisonjoseph @andreancardona @francinelucca @jdharvey-ibm

**Comment by mattrosno** [comment]
@carbon-design-system/carbon-platform-devs I see the web app is now using Next `^12.3.2`. It looks like we're no longer blocked to move scripts to a `_document.js` file if that's needed to close this issue.

Or, maybe we wait until Next 13 to take advantage of the new `app` directory?

https://beta.nextjs.org/docs/upgrade-guide#migrating-from-pages-to-app

https://beta.nextjs.org/docs/routing/fundamentals

**Comment by francinelucca** [comment]
I can take this and move the script so we can close this. Unless we want to hold off until we move to 13 and move to `/app` directory? It's currently in beta and they don't recommend using in `production`... CC: @jdharvey-ibm 
https://beta.nextjs.org/docs/upgrade-guide#migrating-from-pages-to-app
<img width="739" alt="image" src="https://user-images.githubusercontent.com/40550942/204574759-fcea5216-3ecf-4712-ad9b-d4f2f05391a6.png">

**Comment by mattrosno** [comment]
Staying on 12 makes sense to me.

**Comment by jdharvey-ibm** [comment]
We definitely want to move to 13 soon since we have a dependabot PR now for that, but I agree that we can hold off with using the `app` folder for now. Should we use this issue to track the document.js changes *as well as* the effort to migrate to v13?

**Comment by jdharvey-ibm** [comment]
Could also close out this one and open a new issue to track the migration effort from 12 to 13.

**Comment by jdharvey-ibm** [comment]
https://nextjs.org/docs/upgrading

Actually looks like we're blocked there until Carbon supports React 18.

**Comment by francinelucca** [comment]
Worked on this a bit before EOY and was having trouble loading the script from `_document.js` , need to look further into it
